src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/high-risk-data-breaches/HighRiskBreachLayout.tsx

```diff
@@ -57,8 +62,61 @@ export function HighRiskBreachLayout(props: HighRiskBreachLayoutProps) {
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content, exposedData, type } = pageData!;
-  const hasBreaches = type !== "none";
+  const isHighRiskBreachesStep = type !== "none";
   const isStepDone = type === "done";
+  const hasExposedData = exposedData.length > 0;
+
+  // TODO: Write unit tests MNTOR-2560
+  /* c8 ignore start */
+  const handlePrimaryButtonPress = async () => {
+    const highRiskBreachClasses: Record<
+      HighRiskBreachTypes,
+      (typeof HighRiskDataTypes)[keyof typeof HighRiskDataTypes] | null
+    > = {
+      ssn: HighRiskDataTypes.SSN,
+      "credit-card": HighRiskDataTypes.CreditCard,
+      "bank-account": HighRiskDataTypes.BankAccount,
+      pin: HighRiskDataTypes.PIN,
+      none: null,
+      done: null,
+    };
+
+    const dataType = highRiskBreachClasses[type];
+    // Only attempt to resolve the breaches if the following conditions are true:
+    // - There is a matching data class type in this step
+    // - The current step has unresolved exposed data
+    // - There is no pending breach resolution request
+    if (!dataType || !hasExposedData || isResolving) {
+      return;
+    }
+
+    setIsResolving(true);
+    try {
+      const response = await fetch("/api/v1/user/breaches/bulk-resolve", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ dataType }),
+      });
+
+      const result = await response.json();
+      if (!result?.success) {
+        throw new Error(
+          `Could not resolve breach data class of type: ${props.type}`,
+        );
+      }
+
+      const isCurrentStepSection = Object.values(stepMap).includes(nextStep.id);
+      const nextRoute = isCurrentStepSection
+        ? nextStep.href
+        : "/redesign/user/dashboard/fix/high-risk-data-breaches/done";
+      router.push(nextRoute);
+    } catch (_error) {
+      setIsResolving(false);
```

Comment:
Q: Should we send this error to Sentry? Curious what the flow is if the user gets to this catch block (either by invalid async action above, or `!result.success` from the PUT action above. Do they just get caught in a loop of being unable to resolve for infinity if there data is in "a state" or is this only likely to happen on some intermittent server issues where the PUT might randomly be rejected but then fine if they reclick a button?

Reply:
That is an open question I have for @mansaj: I was wondering if not all `throw new Error` calls would be captured by Sentry with `@sentry/nextjs` being setup. We currently don’t have any explicit exceptions being thrown from the frontend anywhere. If that is not the case I’ll definitely add this capture those!

Users will be always able to skip to the next step or re-attempt to resolve a breach step, so they won’t get stuck.

Reply:
Created an issue to make sure we decide on how to handle client error with Sentry and added as a discussion item to our next tech meeting: [fd8634a](https://github.com/mozilla/blurts-server/pull/3791/commits/fd8634a4de4b02e56e498acf08fc0218cb4faafd).